accepted/0000-workspace-run-commands.md

```diff
@@ -4,53 +4,128 @@
+
+### `--parallel`
+
+If the `--parallel` flag is passed, it will runs the commands in parallel in separate child processes, instead of running them in series. This command will ignore the concurrency flag and topological sorting requirements. (Just like Lerna.)
+
+## `yarn workspaces exec ...`
+
+This commands runs an arbitrary shell command in each package. It is similar to `yarn workspaces run`, and respects the same flags, but instead of running a package script defined in `package.json` you can pass it arbitrary shell commands.
+
+This is helpful for cases where you want to execute something that isn't worthy of storing in `package.json`, often when debugging or running a one-off.
+
+For example:
+
+```
+$ yarn workspaces exec babel --out-dir ./lib ./src
+```
+
+## Common Flags
+
+### `--packages`
+
+The `--packages` flag takes a package name or glob, and it restricts the command being run to only take affect in those packages. 
+
+**Note:** this flag operates on the package names, as defined by the `name` field in `package.json` files.
+
+For example:
+
+```
+$ yarn workspaces list --packages 'babel-*' build
+```
+
+
+### `--workspaces`
+
+The `--workspaces` flag takes a workspace name or glob, and it restricts the command being run to only take affect in those workspaces. 
+
+**Note:** this flag operates on the workspace names, not the package names. For example `packages/my-package` would be a workspace name. This is helpful when working with multiple directories of workspaces.
+
+For example, with a `workspaces` setup of:
+
+```json
+[
+  "packages/*",
+  "services/*",
+  "utils/*"
+]
+```
+```
+packages/
+  package-a/
+  package-b/
+  ...
+services/
+  api/
+  cdn/
+utils/
+  ...
+```
+
+You could restrict the command to only run in `services/api` and `services/cdn by doing:
+
+```
+$ yarn workspaces run --workspaces 'services/*' start
+```
 
 # How We Teach This
 
-Since this is a new feature and doesn't affect existing functionalities, it shouldn't affect existing/new users. However, a slight documentation should be written under the workspace CLI.
+Since this is a new feature and doesn't affect existing functionality, it won't change any existing user behavior. However, documentation should be added to explain the new features to those who want to opt-in to them.
 
 # Drawbacks
 
-Complexity. Especially while executing the package scripts. We need to keep track of the package dependencies. And also, there's already a library (lerna) does the same thing and is compatible with yarn without any extra configuration.
+This will increase the complexity of Yarn, instead of letting it live in Lerna and work in tandem. Especially while executing the package scripts, because we need to keep track of the package dependencies. 
 
 # Alternatives
 
-Don't have an alternative way (I can think of) inside yarn.
+There's no current alternative using Yarn. You have to use Lerna and Yarn in concert, which causes confusion as to which one to use when.
```

Comment:
We made https://github.com/whoeverest/wsrun to address this, so I suppose that counts as an alternative.